rustls/src/msgs/handshake.rs

```diff
@@ -219,15 +219,114 @@ impl TlsListElement for SignatureScheme {
     const SIZE_LEN: ListLength = ListLength::U16;
 }
 
+/// Lossy decoding for a `ServerName` extension to a single `DnsName`
 #[derive(Clone, Debug)]
-pub(crate) enum ServerNamePayload {
+pub(crate) enum LossyDnsName<'a> {
```

Comment:
Don't love this name -- how about something like `ServerNameCandidate`?

Reply:
This fell away and remains `ServerNamePayload`